librad/src/net/protocol.rs

```diff
@@ -193,6 +193,11 @@ where
         self.gossip.announce(have).await
     }
 
+    /// Returns `true` if there is at least one active connection.
+    pub async fn is_connected(&self) -> bool {
```

Comment:
What do we think about giving it a parameter for a threshold? Like we only consider ourselves connected if `x` connections are found?

Reply:
I'd like to hear a use-ase for that requirement. Because if we go that specific we could alternatiely just implement a method which returns the number of currently active connections.

Reply:
Ya, `num_connected -> usize` may also be useful.

Reply:
Happy to change it, if we think it's the better API. @kim @NunoAlexandre Any strong feelings on this?

Reply:
> may also be useful.

Could do both, indeed :)

Reply:
How excited are we about 749b657?

Reply:
In the spirit of Rust, both

Reply:
"Clippy: is_empty should be implemented with len"